### PR TITLE
refactor: use u64 instead of U256

### PR DESCRIPTION
**Pull request by Wodann**
While working on this, I also had to fix some errors in the JSON-RPC client APIs

### CODE REVIEW

File: crates/edr_eth/src/remote/client.rs [hunk 1]
```diff
@@ -636,24 +637,26 @@ impl RpcClient {
...
     /// Calls `eth_chainId` and returns the chain ID.
-    pub async fn chain_id(&self) -> Result<U256, RpcClientError> {
+    pub async fn chain_id(&self) -> Result<u64, RpcClientError> {
         let chain_id = *self
             .chain_id
             .get_or_try_init(|| async {
-                self.call_without_cache(MethodInvocation::ChainId()).await
+                self.call_without_cache::<U64>(MethodInvocation::ChainId())
+                    .await
+                    .map(|chain_id| chain_id.as_limbs()[0])
```
Comment (agostbiro) [inline crates/edr_eth/src/remote/client.rs:659]: I think `try_into` should work here.

File: crates/edr_eth/src/remote/client.rs [hunk 2]
```diff
@@ -1753,13 +1745,13 @@ mod tests {
                 .expect("failed to parse address");
 
             let error = TestRpcClient::new(&alchemy_url)
```
Comment (agostbiro) [inline crates/edr_eth/src/remote/client.rs:1747]: I feel like we could just delete this test. As far as I understand, it's testing for a specific error response from Alchemy, which is liable to change.

**Review by agostbiro — CHANGES_REQUESTED** [review]

File: crates/edr_eth/src/remote/client.rs (hunk 1 above)
Comment (Wodann) [reply crates/edr_eth/src/remote/client.rs:659]: It's infallible though. So we'd be unnecessarily wrapping it in a result and then calling expect

File: crates/edr_eth/src/remote/client.rs (hunk 2 above)
Comment (Wodann) [reply crates/edr_eth/src/remote/client.rs:1747]: Do you mean to remove the whole test or the specific error?

I feel that the tests did provide value in this case, as some APIs changed from returning an error to returning an option. Having the tests, helped us catch that instead.

Comment (agostbiro) [reply crates/edr_eth/src/remote/client.rs:1747]: Ah ok I didn't catch that 👍 I'd probably remove the check for a specific error code or message, but I'll leave it up to you.

**Review by agostbiro — APPROVED** [review]

**Pull request merged**